src/mailbox.cpp

```diff
@@ -6,124 +6,107 @@
 #include "mailbox.h"
 
 #include "game.h"
-#include "inbox.h"
 #include "iologindata.h"
 
 extern Game g_game;
 
-ReturnValue Mailbox::queryAdd(int32_t, const Thing& thing, uint32_t, uint32_t, Creature*) const
-{
-	const Item* item = thing.getItem();
-	if (item && Mailbox::canSend(item)) {
-		return RETURNVALUE_NOERROR;
-	}
-	return RETURNVALUE_NOTPOSSIBLE;
-}
-
-ReturnValue Mailbox::queryMaxCount(int32_t, const Thing&, uint32_t count, uint32_t& maxQueryCount, uint32_t) const
-{
-	maxQueryCount = std::max<uint32_t>(1, count);
-	return RETURNVALUE_NOERROR;
-}
+namespace {
 
-ReturnValue Mailbox::queryRemove(const Thing&, uint32_t, uint32_t, Creature* /*= nullptr */) const
+bool canSend(const std::shared_ptr<const Item>& item)
 {
-	return RETURNVALUE_NOTPOSSIBLE;
+	return item->getID() == ITEM_PARCEL || item->getID() == ITEM_LETTER;
 }
 
-Cylinder* Mailbox::queryDestination(int32_t&, const Thing&, Item**, uint32_t&) { return this; }
-
-void Mailbox::addThing(Thing* thing) { return addThing(0, thing); }
-
-void Mailbox::addThing(int32_t, Thing* thing)
+std::string getReceiver(const std::shared_ptr<Item>& item)
```

Comment:
We could rename it to `getReceiverName` to avoid confusion with `getReceiver` method.